pytest.ini

```diff
@@ -1,10 +1,8 @@
 [pytest]
-addopts = -p no:legacypath --color=yes --durations=10 --showlocals -v
+addopts = -p no:legacypath --strict-markers --color=yes --durations=10 --showlocals -v
```

Comment:
The original motivation was to suppress the following warning:

https://github.com/mlflow/mlflow/actions/runs/7842523423/job/21401028272

```
  /home/runner/work/mlflow/mlflow/tests/models/test_cli.py:79: PytestUnknownMarkWarning: Unknown pytest.mark.allow_infer_pip_requirements_fallback - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
```